source/StellarObject.cpp

```diff
@@ -145,3 +180,85 @@ double StellarObject::Distance() const
 {
 	return distance;
 }
+
+
+
+// Selects the nearest potential target and fires projectiles.
+void StellarObject::TryToFire(vector<Projectile> &projectiles, const System *system, vector<Visual> &visuals, const list<std::shared_ptr<Ship>> &ships) const
+{
+	if(!launcher)
+		return;
+	const Government *gov = planet ? planet->GetGovernment() : system->GetGovernment();
+	if(!gov)
+		return;
+	if(!rotation && Random::Real() > reload)
```

Comment:
You've got 3 checks for rotation and 2 checks for a random greater than `reload`, and 0 code comments to signal the intent of these checks. Smells.